git-cliff-core/src/repo.rs

```diff
@@ -510,6 +518,73 @@ impl Repository {
     }
 }
 
+/// Compares two version strings semantically.
+///
+/// This function attempts to parse version strings and compare them semantically
+/// rather than alphabetically. It handles common version formats like:
+/// - v1.2.3
+/// - 1.2.3
+/// - v1.2.3-alpha.1
```

Comment:
nit:

```suggestion
/// This function attempts to parse version strings and compare them semantically
/// rather than alphabetically. It handles common version formats like:
///
/// - v1.2.3
/// - 1.2.3
/// - v1.2.3-alpha.1
```